iOSNativeAOT app build failing

### Description

The build of the HelloiOS Native AOT app is failing in the performance pipeline on latest runtime: https://dev.azure.com/dnceng/internal/_build/results?buildId=2206107 -> ios-arm64 release iOSNativeAOT

### Reproduction Steps

```
# Build the runtime product (possibly from runtime/eng or runtime)
./build.sh -ci -arch arm64 -os ios /p:BuildDarwinFrameworks=true
# Make the app from runtime/src/mono/sample/iOS-NativeAOT
make world TARGET_OS=ios TARGET_ARCH=arm64 BUILD_CONFIG=Release DEPLOY_AND_RUN=false STRIP_DEBUG_SYMBOLS=false
```

### Expected behavior

App succeeds in being built.

### Actual behavior

Build fails with:
```
Time Elapsed 00:16:30.75
/Users/runner/work/1/s/dotnet.sh build -c Release /Users/runner/work/1/s/src/tasks/AppleAppBuilder/AppleAppBuilder.csproj
MSBuild version 17.7.0-preview-23251-02+59879b095 for .NET
  Determining projects to restore...
  Restored /Users/runner/work/1/s/src/tasks/AppleAppBuilder/AppleAppBuilder.csproj (in 368 ms).
  AppleAppBuilder -> /Users/runner/work/1/s/artifacts/bin/AppleAppBuilder/Release/net8.0/AppleAppBuilder.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:06.34
rm -rf obj bin *.binlog 
/Users/runner/work/1/s/dotnet.sh \
	publish -c Release \
	-p:TargetOS=ios \
	-p:TargetArchitecture=arm64 \
	-p:DeployAndRun=false \
	-p:StripDebugSymbols=false \
	-p:PublishAotUsingRuntimePack=false \
	-bl
MSBuild version 17.7.0-preview-23251-02+59879b095 for .NET
  Determining projects to restore...
  Restored /Users/runner/work/1/s/src/mono/sample/iOS-NativeAOT/Program.csproj (in 373 ms).
  Program -> /Users/runner/work/1/s/src/mono/sample/iOS-NativeAOT/bin/ios-arm64/Program.dll
  Generating native code
/Users/runner/work/1/s/src/mono/sample/iOS-NativeAOT/Program.csproj(72,50): error MSB4184: The expression "[System.String]::new(''-Wl, -rpath, @executable_path'')" cannot be evaluated. Attempted to access a missing method.
make: *** [hello-app] Error 1
##[error]Bash exited with code '2'.

```

### Regression?

Was working with runtime version 1ff48ad53e01d1db3b5ad6b247347dc5609cb93a, first failure with version fe88013164abfd9cfb3e1dc3b518acf4c7c924a9.

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

_No response_

--- a/src/mono/sample/iOS-NativeAOT/Program.csproj
+++ b/src/mono/sample/iOS-NativeAOT/Program.csproj
@@ -69,7 +69,6 @@
 
     <ItemGroup>
       <_LinkerFlagsToDrop Include="@(NativeFramework->'-framework %(Identity)')" />
-      <_LinkerFlagsToDrop Include="@(LinkerArg)" Condition="$([System.String]::new('%(Identity)').Contains('swift'))" />
       <LinkerArg Remove="@(_LinkerFlagsToDrop)" />
       <ExtraLinkerArguments Include="@(LinkerArg)" />
       <_NativeDependencies Include="%(ManagedBinary.IlcOutputFile)" />

